### ISSUE TITLE
CMake Error at cmake/OpenCVCompilerOptimizations.cmake:611 - Compiler doesn't support baseline optimization flags

### ISSUE THREAD

**Issue by anandb7248** [opencv/opencv#23396]
### System Information

Python version: 3.11.2
OpenCV version: 4.x 
Mac version: Monterey 12.6 
XCode version: 14.2 

### Detailed description

Have a React Native project integrating with OpenCV where the simulator isn't working due to architecture difference. https://github.com/opencv/opencv/issues/21926 

Then tried to follow the steps in:
https://github.com/opencv/opencv/tree/4.x/platforms/ios
https://github.com/opencv/opencv/tree/4.x/platforms/apple

But when trying to run:
python opencv/platforms/apple/build_xcframework.py --out ./build_xcframework

Keep getting an error:
```
CMake Error at cmake/OpenCVCompilerOptimizations.cmake:611 (message):
  Compiler doesn't support baseline optimization flags:
Call Stack (most recent call first):
  cmake/OpenCVCompilerOptions.cmake:327 (ocv_compiler_optimization_options)
  CMakeLists.txt:650 (include)
```

### Steps to reproduce

``` 
git clone https://github.com/opencv/opencv.git 

Then try to follow the steps in:
https://github.com/opencv/opencv/tree/4.x/platforms/ios
https://github.com/opencv/opencv/tree/4.x/platforms/apple
```

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

**Comment by pdavid0** [comment]
Same here, there's no error message other than "Non Zero Exit Code" 

```
-- General configuration for OpenCV 4.7.0-dev =====================================
--   Version control:               4.7.0-137-g352f92e437
--
--   Extra modules:
--     Location (extra):            /Users/dev/workspace/opencv_contrib/modules
--     Version control (extra):     4.7.0-27-g6ebb38db
--
--   Platform:
--     Timestamp:                   2023-03-24T20:01:38Z
--     Host:                        Darwin 22.1.0 arm64
--     Target:                      iOS 9.0 arm64
--     CMake:                       3.26.1
--     CMake generator:             Xcode
--     CMake build tool:            /Users/dev/workspace/ios/build/build-arm64-iphoneos/xcodebuild_wrapper
--     Xcode:                       14.2
--     Configuration:               Debug Release
--
--   CPU/HW features:
--     Baseline:                    NEON FP16
--       requested:                 DETECT
--
--   C/C++:
--     Built as dynamic libs?:      NO
--     C++ standard:                11
--     C++ Compiler:                /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++  (ver 14.0.0.14000029)
--     C++ flags (Release):         -fembed-bitcode   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Qunused-arguments -Wno-semicolon-before-method-body  -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated-copy -O3 -DNDEBUG  -DNDEBUG
--     C++ flags (Debug):           -fembed-bitcode   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Qunused-arguments -Wno-semicolon-before-method-body  -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated-copy -g  -O0 -DDEBUG -D_DEBUG
--     C Compiler:                  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang
--     C flags (Release):           -fembed-bitcode   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Qunused-arguments -Wno-semicolon-before-method-body  -fvisibility=hidden -fvisibility-inlines-hidden -O3 -DNDEBUG  -DNDEBUG
--     C flags (Debug):             -fembed-bitcode   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Qunused-arguments -Wno-semicolon-before-method-body  -fvisibility=hidden -fvisibility-inlines-hidden -g  -O0 -DDEBUG -D_DEBUG
--     Linker flags (Release):
--     Linker flags (Debug):
--     ccache:                      NO
--     Precompiled headers:         NO
--     Extra dependencies:          -framework UIKit -framework Accelerate -framework CoreGraphics -framework QuartzCore Iconv::Iconv
--     3rdparty dependencies:       libprotobuf libjpeg-turbo libwebp libpng zlib quirc ade
--
--   OpenCV modules:
--     To be built:                 aruco barcode bgsegm bioinspired calib3d ccalib core datasets dnn dnn_objdetect dnn_superres dpm face features2d flann fuzzy gapi hfs highgui img_hash imgcodecs imgproc intensity_transform line_descriptor mcc ml objc objdetect optflow phase_unwrapping photo plot quality rapid reg rgbd saliency shape stereo stitching structured_light surface_matching text tracking video videoio videostab wechat_qrcode world xfeatures2d ximgproc xobjdetect xphoto
--     Disabled:                    -
--     Disabled by dependency:      -
--     Unavailable:                 alphamat cudaarithm cudabgsegm cudacodec cudafeatures2d cudafilters cudaimgproc cudalegacy cudaobjdetect cudaoptflow cudastereo cudawarping cudev cvv freetype hdf java julia matlab ovis python2 python3 sfm superres ts viz
--     Applications:                -
--     Documentation:               NO
--     Non-free algorithms:         NO
--
--   GUI:
--
--   Media I/O:
--     ZLib:                        build (ver 1.2.13)
--     JPEG:                        build-libjpeg-turbo (ver 2.1.3-62)
--       SIMD Support Request:      YES
--       SIMD Support:              YES
--     WEBP:                        build (ver encoder: 0x020f)
--     PNG:                         build (ver 1.6.37)
--     HDR:                         YES
--     SUNRASTER:                   YES
--     PXM:                         YES
--     PFM:                         YES
--
--   Video I/O:
--     AVFoundation:                YES
--     iOS capture:                 YES
--
--   Parallel framework:            GCD
--
--   Trace:                         YES (built-in)
--
--   Other third-party libraries:
--     Custom HAL:                  NO
--     Protobuf:                    build (3.19.1)
--     Flatbuffers:                 builtin/3rdparty (23.1.21)
--
--   Python (for build):            NO
--
--   Objective-C wrappers:          YES
--
--   Install to:                    /Users/dev/workspace/ios/build/build-arm64-iphoneos/install
-- -----------------------------------------------------------------
--
-- Configuring incomplete, errors occurred!
============================================================
ERROR: Command '['cmake', '-GXcode', '-DAPPLE_FRAMEWORK=ON', '-DCMAKE_INSTALL_PREFIX=install', '-DCMAKE_BUILD_TYPE=Release', '-DOPENCV_INCLUDE_INSTALL_PATH=include', '-DOPENCV_3P_LIB_INSTALL_PATH=lib/3rdparty', '-DFRAMEWORK_NAME=opencv2', '-DIOS_ARCH=arm64', '-DCMAKE_TOOLCHAIN_FILE=/Users/dev/workspace/opencv/platforms/ios/cmake/Toolchains/Toolchain-iPhoneOS_Xcode.cmake', '-DCPU_BASELINE=DETECT', '/Users/dev/workspace/opencv', '-DOPENCV_EXTRA_MODULES_PATH=/Users/dev/workspace/opencv_contrib/modules', '-DCMAKE_C_FLAGS=-fembed-bitcode', '-DCMAKE_CXX_FLAGS=-fembed-bitcode']' returned non-zero exit status 1.
============================================================
Traceback (most recent call last):
File "/Users/dev/workspace/opencv/platforms/ios/build_framework.py", line 181, in build
self._build(outdir)
File "/Users/dev/workspace/opencv/platforms/ios/build_framework.py", line 139, in _build
self.buildOne(target[0], target[1], main_build_dir, cmake_flags)
File "/Users/dev/workspace/opencv/platforms/ios/build_framework.py", line 306, in buildOne
execute(cmakecmd, cwd = builddir)
File "/Users/dev/workspace/opencv/platforms/apple/cv_build_utils.py", line 13, in execute
retcode = check_call(cmd, cwd = cwd)
File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/subprocess.py", line 369, in check_call
raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '-GXcode', '-DAPPLE_FRAMEWORK=ON', '-DCMAKE_INSTALL_PREFIX=install', '-DCMAKE_BUILD_TYPE=Release', '-DOPENCV_INCLUDE_INSTALL_PATH=include', '-DOPENCV_3P_LIB_INSTALL_PATH=lib/3rdparty', '-DFRAMEWORK_NAME=opencv2', '-DIOS_ARCH=arm64', '-DCMAKE_TOOLCHAIN_FILE=/Users/dev/workspace/opencv/platforms/ios/cmake/Toolchains/Toolchain-iPhoneOS_Xcode.cmake', '-DCPU_BASELINE=DETECT', '/Users/dev/workspace/opencv', '-DOPENCV_EXTRA_MODULES_PATH=/Users/dev/workspace/opencv_contrib/modules', '-DCMAKE_C_FLAGS=-fembed-bitcode', '-DCMAKE_CXX_FLAGS=-fembed-bitcode']' returned non-zero exit status 1.
```

**Comment by m-massaro** [comment]
I am facing the same error while trying to build the xcframework to support multiple architecture with this command:
```
python3 opencv/platforms/apple/build_xcframework.py \
--out ./opencv-build \
--contrib opencv_contrib \
--iphoneos_archs armv7,armv7s,arm64 \
--iphonesimulator_archs arm64 \
--build_only_specified_archs
```

The error is:
```
=================================
CMake
=================================

Executing: ['cmake', '-GXcode', '-DAPPLE_FRAMEWORK=ON', '-DCMAKE_INSTALL_PREFIX=install', '-DCMAKE_BUILD_TYPE=Release', '-DOPENCV_INCLUDE_INSTALL_PATH=include', '-DOPENCV_3P_LIB_INSTALL_PATH=lib/3rdparty', '-DFRAMEWORK_NAME=opencv2', '-DIOS_ARCH=armv7', '-DCMAKE_TOOLCHAIN_FILE=/Users/michele/Development/software/opencv/opencv/platforms/ios/cmake/Toolchains/Toolchain-iPhoneOS_Xcode.cmake', '-DCPU_BASELINE=DETECT', '/Users/michele/Development/software/opencv/opencv', '-DOPENCV_EXTRA_MODULES_PATH=/Users/michele/Development/software/opencv/opencv_contrib/modules', '-DCMAKE_C_FLAGS=-fembed-bitcode', '-DCMAKE_CXX_FLAGS=-fembed-bitcode'] in /Users/michele/Development/software/opencv/opencv-build/iphoneos/build/build-armv7-iphoneos
Executing: cmake -GXcode -DAPPLE_FRAMEWORK=ON -DCMAKE_INSTALL_PREFIX=install -DCMAKE_BUILD_TYPE=Release -DOPENCV_INCLUDE_INSTALL_PATH=include -DOPENCV_3P_LIB_INSTALL_PATH=lib/3rdparty -DFRAMEWORK_NAME=opencv2 -DIOS_ARCH=armv7 -DCMAKE_TOOLCHAIN_FILE=/Users/michele/Development/software/opencv/opencv/platforms/ios/cmake/Toolchains/Toolchain-iPhoneOS_Xcode.cmake -DCPU_BASELINE=DETECT /Users/michele/Development/software/opencv/opencv -DOPENCV_EXTRA_MODULES_PATH=/Users/michele/Development/software/opencv/opencv_contrib/modules -DCMAKE_C_FLAGS=-fembed-bitcode -DCMAKE_CXX_FLAGS=-fembed-bitcode
-- Setting up iPhoneOS toolchain for IOS_ARCH='armv7'
-- iPhoneOS toolchain loaded
-- Detected processor: armv7
-- Looking for ccache - not found
-- VFPV3 is not supported by C++ compiler
-- NEON is not supported by C++ compiler
-- FP16 is not supported by C++ compiler
CMake Error at cmake/OpenCVCompilerOptimizations.cmake:611 (message):
  Compiler doesn't support baseline optimization flags:
Call Stack (most recent call first):
  cmake/OpenCVCompilerOptions.cmake:327 (ocv_compiler_optimization_options)
  CMakeLists.txt:650 (include)
```

**Comment by fnc12** [comment]
Same issue.

Python version: 3.11.3
OpenCV version: 3.4.16
macOS version: 13.2.1
Xcode version: 14.2

**Comment by fnc12** [comment]
@krjw-eyev did not change anything for me =(

**Comment by aschweiz** [comment]
> @krjw-eyev did not change anything for me =(

Removing the armv7 and armv7s target architectures and keeping only arm64 worked for me.

My setup:
Apple clang version 14.0.3 (clang-1403.0.22.14.1)
Target: x86_64-apple-darwin22.4.0
Thread model: posix

**Comment by Percylau120416** [comment]
How to remove armv7 and armv7s target arch?

**Comment by MarkBirch3D** [comment]
I am also having the same issue. Trying to build OpenCV as xcframework so we can run our app in simulator, but i am having the issue above.

Python: 3.11.4
OpenCV: 4.x
macOS: M1 Ventura 13.4.1
XCode: 14.3

**Comment by Antsthebul** [comment]
Mac OS: 12.67
Make: 3.27.4
python: 3.11/2.7
Xcode: 14.2
Desired build for iOS:16

Attempting to build for iOs. 
Attempt 1: 
Failed while building for ARM7 with the above error code

Attempt 2:
I built for ARM64 only . This can be done by specifying environment/runtime variables `python opencv/platforms/ios/build_framework.py ios  --iphoneos_archs arm64 ` or (what I did)removing the architectures in https://github.com/opencv/opencv/blob/4.x/platforms/ios/build_framework.py#L555

This also failed because of the Xcode version was targeting platforms 11-16.2.9, so I also hardcoded change here https://github.com/opencv/opencv/blob/4.x/platforms/ios/build_framework.py#L47 

Attempt 3:
Still failed but was either along and so I downloaded and switched to Python2.7

Solution:
Attempt3 made it the furthest and still failed with the above error message.  I downloaded the necessary files from https://opencv.org/releases/ under the iOS pack

**Comment by niqt** [comment]
Any news about this issue?

**Comment by trunghvbk** [comment]
> I am also having the same issue. Trying to build OpenCV as xcframework so we can run our app in simulator, but i am having the issue above.
> 
> Python: 3.11.4 OpenCV: 4.x macOS: M1 Ventura 13.4.1 XCode: 14.3 Clang: 14.0.3
> 
> EDIT: I tried removing armv7 as aschweiz suggested and I get much further into the script, but it still fails later on
> 
> ```
> ** BUILD FAILED **
> 
> The following build commands failed:
> 	PhaseScriptExecution Generate\ CMakeFiles/dephelper/gen_opencv_objc_source_ios /Users/mab/build_xcframework/iphoneos/build/build-arm64-iphoneos/build/OpenCV.build/Release-iphoneos/gen_opencv_objc_source_ios.build/Script-F1D1F124C291ED2EE5DE0A39.sh (in target 'gen_opencv_objc_source_ios' from project 'OpenCV')
> (1 failure)
> ============================================================
> ERROR: Command '['xcodebuild', 'BITCODE_GENERATION_MODE=bitcode', 'IPHONEOS_DEPLOYMENT_TARGET=9.0', 'ARCHS=arm64', '-sdk', 'iphoneos', '-configuration', 'Release', '-parallelizeTargets', '-jobs', '8', '-target', 'ALL_BUILD', 'build']' returned non-zero exit status 65.
> ============================================================
> Traceback (most recent call last):
>   File "/Users/mab/opencv/platforms/ios/build_framework.py", line 181, in build
>     self._build(outdir)
>   File "/Users/mab/opencv/platforms/ios/build_framework.py", line 139, in _build
>     self.buildOne(target[0], target[1], main_build_dir, cmake_flags)
>   File "/Users/mab/opencv/platforms/ios/build_framework.py", line 318, in buildOne
>     execute(buildcmd + ["-target", "ALL_BUILD", "build"], cwd = builddir)
>   File "/Users/mab/opencv/platforms/apple/cv_build_utils.py", line 13, in execute
>     retcode = check_call(cmd, cwd = cwd)
>               ^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 413, in check_call
>     raise CalledProcessError(retcode, cmd)
> subprocess.CalledProcessError: Command '['xcodebuild', 'BITCODE_GENERATION_MODE=bitcode', 'IPHONEOS_DEPLOYMENT_TARGET=9.0', 'ARCHS=arm64', '-sdk', 'iphoneos', '-configuration', 'Release', '-parallelizeTargets', '-jobs', '8', '-target', 'ALL_BUILD', 'build']' returned non-zero exit status 65.
> ============================================================
> ERROR: Command '['python3', '/Users/mab/opencv/platforms/ios/build_framework.py', './build_xcframework/iphoneos', '--iphoneos_archs', 'arm64', '--framework_name', 'opencv2', '--build_only_specified_archs']' returned non-zero exit status 1.
> ============================================================
> Traceback (most recent call last):
>   File "/Users/mab/opencv/platforms/apple/build_xcframework.py", line 87, in <module>
>     execute(command, cwd=os.getcwd())
>   File "/Users/mab/opencv/platforms/apple/cv_build_utils.py", line 13, in execute
>     retcode = check_call(cmd, cwd = cwd)
>               ^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 413, in check_call
>     raise CalledProcessError(retcode, cmd)
> subprocess.CalledProcessError: Command '['python3', '/Users/mab/opencv/platforms/ios/build_framework.py', './build_xcframework/iphoneos', '--iphoneos_archs', 'arm64', '--framework_name', 'opencv2', '--build_only_specified_archs']' returned non-zero exit status 1.
> ```

Mine is the same.
MacOS: 13.4.1
Xcode: 14.3
openCV: 4.8.0
python: 3.10.4

**Comment by Fando** [comment]
## TLDR
Compiling OpenCV framework for iOS worked for me by limiting it to `arm64` and `x86_64` architectures. The resulting framework works when running the app on a physical device, but not on a simulator.

## Error
Including `armv7` or `armv7s` architectures, or `i386` simulator architecture, is causing the source compilation to fail with the following error:
```
CMake Error at cmake/OpenCVCompilerOptimizations.cmake:611 (message):
  Compiler doesn't support baseline optimization flags:
Call Stack (most recent call first):
  cmake/OpenCVCompilerOptions.cmake:327 (ocv_compiler_optimization_options)
  CMakeLists.txt:652 (include)
```

## System Settings
- ios deployment target: 14.0
- opencv: 4.8.0
- macos: 13.4.1 (c) 
- chip: Apple M1 Pro
- xcode: 14.3.1 
- python:  3.11.6
- cmake: 3.27.7

## Success
I was able to compile successfully by doing the following

- Remove all iphone architectures except `arm64` using the `iphoneos_archs` flag
- Remove all simulator architectures except `x86_64` using the `iphonesimulator_archs` flag

I also did the following, which shouldn't impact the compilation, but I mention it for completeness. (I think it may be important to set target to 14 or greater, but not sure)
- Set target to `14.0` using `iphoneos_deployment_target` flag 
- I also excluded all modules except `core`, `imgcodecs`, `imgproc` because I only need these 3

## Successful build command
```
python3 build_framework.py ios 
--iphoneos_deployment_target "14.0" 
--build_only_specified_archs 
--iphoneos_archs "arm64" 
--iphonesimulator_archs "x86_64"
--without calib3d 
--without dnn 
--without features2d 
--without flann
--without gapi 
--without highgui 
--without java 
--without js
--without ml
--without objc 
--without objdetect 
--without photo 
--without python 
--without stitching 
--without ts
--without video 
--without videoio 
--without world 
```

## Works on Physical Device
Adding the framework to the project and building/running the app on a physical device works, `iPhone 8`, `iOS 16.3`

## Error on Simulator
However when trying to build and run the app on a simulator, `iPhone 14 Pro`, it fails with these 2 errors:
```
ld: in .../opencv2.framework/opencv2(algorithm.o), building for iOS Simulator, 
but linking in object file built for iOS, file '.../opencv2.framework/opencv2' for architecture arm64

clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

## Helpful Resource
[iOS Devices: Releases, Firmware, Instruction Sets, Screen Sizes](https://www.innerfence.com/howto/apple-ios-devices-dates-versions-instruction-sets)

**Comment by gl-lei** [comment]
Upgrading CMake to the latest version solved this problem (cmake version 3.28.2)

**Comment by XuYoujin** [comment]
who success? tell us

**Comment by aleksandaratanasov** [comment]
I just had this problem while setting up a CI job in GitLab. Initially working all of a sudden I started getting this error after restructuring my CI config files and especially the one that was calling the `cmake` configuration and build.

Currently I use `CI_COMMIT_REF_NAME`, which is rename to `release` in case the commit is coming from the `main`  branch. First I started getting a warning that OpenCV will use `Release` (notice the capital `R`) as default and I need to specify `CMAKE_BUILD_TYPE` even though I was specifying `CMAKE_CONFIGURATION_TYPES` since I am using MSVC. In order to actually make things work I added `CMAKE_BUILD_TYPE` but I forgot the quotation marks. O_o

I took me two hours to figure out that I had to use `"release"` (or whatever the build type) for `-DCMAKE_BUILD_TYPE`.

The respective sections of my CI are listed below:

```
.rename-main-commit-ref:
  script:
    - |
      # main == release but we have to rename it to keep it consistent
      # This check should be dropped in the future (perhaps) for the tag
      # check and proper release (special type of CI job)
      if($CI_COMMIT_REF_NAME -eq "main"){
        echo "Found 'main' commit ref, will rename to 'release'"
        $CI_COMMIT_REF_NAME = "release"
      }
      else{
        echo "Skipping renaming of commit ref since not 'main'"
      }

.build:
  variables:
    DepName: ""
    DepsRoot: ""
    DepCmakeArg: -DBuildDEFAULT=ON
  script:
    - !reference [.rename-main-commit-ref, script]
    - |
      cmake -Bbuild -G "Visual Studio 16 2019" -S. -A x64 -DCMAKE_BUILD_TYPE="$CI_COMMIT_REF_NAME" -DCMAKE_CONFIGURATION_TYPES="$($CI_COMMIT_REF_NAME)" $DepCmakeArg
      cmake --build build --target ALL_BUILD --config "$CI_COMMIT_REF_NAME"

```

When the error was occurring I had `-DCMAKE_BUILD_TYPE=$CI_COMMIT_REF_NAME`. :D Funny enough, this does not seem to cause issues with other CMake project I have. I believe that there is an issue with the interpretation of `CMAKE_BUILD_TYPE`.

**Comment by kopyl** [comment]
@krjw-eyev could you please help me?

Is there any way i can adjust my building command to fix it the same way as you did?

Here is how i build it
```
IPHONEOS_DEPLOYMENT_TARGET=17.0 cmake \
  -D CMAKE_TOOLCHAIN_FILE=../opencv/platforms/ios/cmake/Toolchains/Toolchain-iPhoneOS_Xcode.cmake \
  -D PLATFORM=OS64 \
  -D IOS_ARCH=arm64 \
  -D BUILD_SHARED_LIBS=OFF \
  -D BUILD_opencv_world=OFF \
  -D OPENCV_MODULES_TO_BUILD=core,imgproc,dnn \
  -D CMAKE_INSTALL_PREFIX=/Users/olehkopyl/Downloads/open-cv/make_install_outputs_ios \
  -D ENABLE_BITCODE=ON \
  -D HAVE_CXX11=ON \
  -D CMAKE_CXX_COMPILER=/usr/bin/clang++ \
  ..
```
